linkerd/proxy/tcp/src/balance.rs

```diff
@@ -1,28 +1,86 @@
 use linkerd_error::Error;
-use linkerd_stack::layer;
+use linkerd_proxy_core::Resolve;
+use linkerd_proxy_discover::{self as discover, NewSpawnDiscover};
+use linkerd_stack::{layer, NewService, Param, Service};
 use rand::thread_rng;
-use std::{hash::Hash, time::Duration};
-pub use tower::{
-    balance::p2c::Balance,
-    load::{Load, PeakEwmaDiscover},
+use std::{marker::PhantomData, net::SocketAddr, time::Duration};
+use tower::{
+    balance::p2c,
+    load::{CompleteOnResponse, PeakEwmaDiscover},
 };
-use tower::{discover::Discover, load::CompleteOnResponse};
-
-/// Produces a PeakEWMA balancer that uses connect latency (and pending
-/// connections) as its load metric.
-pub fn layer<T, D>(
-    default_rtt: Duration,
-    decay: Duration,
-) -> impl tower::layer::Layer<D, Service = Balance<PeakEwmaDiscover<D, CompleteOnResponse>, T>> + Clone
+
+#[derive(Clone, Debug)]
+pub struct EwmaConfig {
+    pub default_rtt: Duration,
+    pub decay: Duration,
+}
+
+/// Configures a stack to resolve `T` typed targets to balance requests over
+/// `M`-typed endpoint stacks.
+#[derive(Debug)]
+pub struct NewBalance<I, R, N> {
+    resolve: R,
+    inner: N,
+    _marker: PhantomData<fn(I)>,
+}
+
+pub type Balance<I, S> = p2c::Balance<PeakEwmaDiscover<discover::Buffer<S>, CompleteOnResponse>, I>;
+
+// === impl NewBalance ===
+
+impl<I, R, N> NewBalance<I, R, N> {
+    // FIXME(ver)
+    const CAPACITY: usize = 1_000;
```

Comment:
can/should this also be on `EwmaConfig`? it's not technically a EWMA parameter, and it's tuning the resolution config rather than the balancer...and i'm not sure if we'd want this to be part of the discovered configuration anyway?